baseframe/forms/validators.py

```diff
@@ -161,6 +162,30 @@ def compare(self, value, other):
         return value != other
 
 
+class PublicEmailDomain(object):
```

Comment:
I think we should make this `IsPublicEmailDomain` since the name is slightly ambiguous.